// pro-ui-library/dist/cjs/pro-card_4.cjs.entry.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

const __chunk_1 = require('./chunk-4c2d6ff3.js');

class ProCard {
    constructor(hostRef) {
        __chunk_1.registerInstance(this, hostRef);
        this.button = false;
        this.type = 'button';
        this.disabled = false;
        this.win = __chunk_1.getContext(this, "window");
    }
    isClickable() {
        return (this.href !== undefined || this.button);
    }
    hostData() {
        return {
            class: {
                'card-disabled': this.disabled,
                'ion-activatable': this.isClickable()
            }
        };
    }
    __stencil_render() {
        const clickable = this.isClickable();
        if (!clickable) {
            return [
                __chunk_1.h("slot", null)
            ];
        }
        const { href } = this;
        const TagType = clickable ? (href === undefined ? 'button' : 'a') : 'div';
        const attrs = (TagType === 'button')
            ? { type: this.type }
            : {
                href: this.href,
                rel: this.rel,
                target: this.target
            };
        return (__chunk_1.h(TagType, Object.assign({}, attrs, { class: "card-native", disabled: this.disabled }), __chunk_1.h("slot", null), clickable && __chunk_1.h("pro-ripple-effect", null)));
    }
    render() { return __chunk_1.h(__chunk_1.Host, this.hostData(), this.__stencil_render()); }
    static get style() { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}pro-card{display:block;position:relative;overflow:hidden;-webkit-box-shadow:0 1px 3px #b7b7b7;box-shadow:0 1px 3px #b7b7b7;-webkit-transition:.3s;transition:.3s;border-color:#0a0908;border-width:1px;font-size:16px;padding:5px;margin:5px;background-color:#fdfffc;min-height:200px;max-height:500px}"; }
}

class ProImage {
    constructor(hostRef) {
        __chunk_1.registerInstance(this, hostRef);
    }
    // @Watch() src
    render() {
        return (__chunk_1.h("img", { src: this.src, alt: this.alt, decoding: "async" }));
    }
    static get style() { return ":host{display:block;-o-object-fit:contain;object-fit:contain}:host img{display:block;min-width:50px;min-height:50px;max-width:150px;max-height:150px;width:100%;height:100%;-o-object-fit:inherit;object-fit:inherit;-o-object-position:inherit;object-position:inherit}"; }
}

class ProLabel {
    constructor(hostRef) {
        __chunk_1.registerInstance(this, hostRef);
    }
    render() {
        return (__chunk_1.h("label", { htmlfor: this.for }, __chunk_1.h("slot", null)));
    }
    static get style() { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}pro-label{font-size:12px;font-style:italic;text-align:center}"; }
}

class ProModal {
    constructor(hostRef) {
        __chunk_1.registerInstance(this, hostRef);
        this.presented = false;
    }
    async present() {
        if (!this.presented)
            this.presented = true;
    }
    async dismiss() {
        if (this.presented)
            this.presented = false;
    }
    render() {
        return (this.presented) ? (__chunk_1.h(__chunk_1.Host, null, __chunk_1.h("pro-backdrop", { onClick: () => this.dismiss() }), __chunk_1.h("div", { class: "modal-window" }, __chunk_1.h("div", { class: "modal-content" }, __chunk_1.h("slot", null))))) : (__chunk_1.h(__chunk_1.Host, null));
    }
    get el() { return __chunk_1.getElement(this); }
    static get style() { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}:host .modal-window{position:absolute;right:0;left:0;top:2%;bottom:10%;margin-right:auto;margin-left:auto;z-index:15;background:#fdfffc;border-radius:25px;width:80%;min-height:40%;max-height:60%;padding:30px;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center}\@media only screen and (max-width:600px){:host .modal-window{width:80%;min-height:50%}}\@media only screen and (min-width:600px){:host .modal-window{width:80%;min-height:50%}}:host .modal-content{width:100%;height:100%}"; }
}

exports.pro_card = ProCard;
exports.pro_image = ProImage;
exports.pro_label = ProLabel;
exports.pro_modal = ProModal;
