// pro-ui-library/dist/esm/legacy/pro-button_6.entry.js
import { r as registerInstance, h, c as getElement, d as createEvent, H as Host } from './chunk-c73a2de7.js';
var ProButton = /** @class */ (function () {
    function ProButton(hostRef) {
        var _this = this;
        registerInstance(this, hostRef);
        this.type = 'button';
        this.disabled = false;
        this.handleClick = function (ev) {
            var form = _this.el.closest('form');
            if (form) {
                ev.preventDefault();
                var fakeButton = document.createElement('button');
                fakeButton.type = _this.type;
                fakeButton.style.display = 'none';
                form.appendChild(fakeButton);
                fakeButton.click();
                fakeButton.remove();
            }
        };
    }
    ProButton.prototype.render = function () {
        return (h("button", { class: "button-native", disabled: this.disabled, type: this.type, onClick: this.handleClick }, h("span", { class: "button-inner" }, h("slot", { name: "icon-only" }), h("slot", { name: "start" }), h("slot", null), h("slot", { name: "end" }))));
    };
    Object.defineProperty(ProButton.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ProButton, "style", {
        get: function () { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}:host .button-native{background-color:#263a3c;padding:5px;border-radius:10%;padding-top:10px;border-radius:5px;color:#fdfffc;font-family:Raleway,Helvetica,sans-serif;font-size:14px;text-transform:uppercase;width:100%;min-height:35px;max-height:70px;-webkit-transition:.5s;transition:.5s;cursor:pointer}:host .button-inner{font-weight:700}:host button:disabled{background-color:grey;color:#fff;cursor:auto;font-size:14px}:host :hover{background-color:#bdeaf9;color:#263a3c;-webkit-transition:.5s;transition:.5s;font-size:16px}:host :focus{outline:none;-webkit-box-shadow:0 1px 3px #b7b7b7;box-shadow:0 1px 3px #b7b7b7}\@media only screen and (max-width:500px){:host .button-native{font-size:10px}:host :hover{font-size:12px}:host button:disabled{font-size:10px}}"; },
        enumerable: true,
        configurable: true
    });
    return ProButton;
}());
var ProCheckbox = /** @class */ (function () {
    function ProCheckbox(hostRef) {
        var _this = this;
        registerInstance(this, hostRef);
        this.disabled = false;
        this.checked = false;
        this.onClick = function () {
            _this.checked = !_this.checked;
        };
        this.proChange = createEvent(this, "proChange", 7);
        this.proStyle = createEvent(this, "proStyle", 7);
    }
    ProCheckbox.prototype.componentWillLoad = function () {
        this.emitStyle();
    };
    ProCheckbox.prototype.checkedChanged = function (isChecked) {
        this.proChange.emit({
            checked: isChecked,
            value: this.value
        });
        this.emitStyle();
    };
    ProCheckbox.prototype.disabledChanged = function () {
        this.emitStyle();
    };
    ProCheckbox.prototype.emitStyle = function () {
        this.proStyle.emit({
            'checkbox-checked': this.checked,
            'interactive-disabled': this.disabled,
            'checkbox-value': this.value
        });
    };
    ProCheckbox.prototype.render = function () {
        return h("input", { type: "checkbox", onClick: this.onClick });
    };
    Object.defineProperty(ProCheckbox.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ProCheckbox, "watchers", {
        get: function () {
            return {
                "checked": ["checkedChanged"],
                "disabled": ["disabledChanged"]
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ProCheckbox, "style", {
        get: function () { return "\@charset \"UTF-8\";\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}input[type=checkbox]:after{background-color:#dcdcdc;content:\"\"}input[type=checkbox]:after,input[type=checkbox]:checked:after{width:15px;height:15px;border-radius:2px;top:-2px;left:-1px;position:relative;display:inline-block;visibility:visible;border:2px solid #fff}input[type=checkbox]:checked:after{content:\"✔\";color:#30bced;text-align:center}input[type=checkbox]:hover:after{background:#1e96ff;-webkit-transition:1s;transition:1s}"; },
        enumerable: true,
        configurable: true
    });
    return ProCheckbox;
}());
var ProInput = /** @class */ (function () {
    function ProInput(hostRef) {
        registerInstance(this, hostRef);
        this.regex = /^(.+)$/;
        this.required = false;
        this.inputChange = createEvent(this, "inputChange", 7);
    }
    ProInput.prototype.componentWillLoad = function () {
        if (this.required) {
            this.reqLabel = 'required';
        }
    };
    ProInput.prototype.onInputChange = function (ev) {
        var value = ev.target.value;
        var validated = false;
        if (value != '') {
            (this.required) ? validated = this.regex.test(value) : validated = true;
            validated ? this.class = 'valid' : this.class = 'invalid';
        }
        else {
            validated = !this.required;
            validated ? this.class = '' : this.class = 'invalid';
        }
        this.inputChange.emit({
            name: this.label,
            value: value,
            valid: validated
        });
    };
    ProInput.prototype.render = function () {
        var _this = this;
        return (h(Host, null, h("label", { htmlFor: this.label, class: this.reqLabel }, this.label), h("input", { onChange: function (ev) { return _this.onInputChange(ev); }, name: this.label, placeholder: this.placeholder, id: this.label, class: this.class })));
    };
    Object.defineProperty(ProInput, "style", {
        get: function () { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}:host{overflow:hidden}:host input{width:100%;height:20px;border-radius:5px;font-family:Raleway,Helvetica,sans-serif;padding-left:5px;margin-top:2px}:host label{width:100%;font-size:10px;font-style:italic;margin-bottom:40px}:host label.required{border-bottom:1px dotted rgba(240,0,0,.3)}:host input:focus{background-color:#bdeaf9;outline:none}:host input:hover{background-color:#bdeaf9;-webkit-transition:1s;transition:1s}:host input.valid{background-color:#c4d2c1}:host input.invalid{background-color:#e09a9f}"; },
        enumerable: true,
        configurable: true
    });
    return ProInput;
}());
var ProList = /** @class */ (function () {
    function ProList(hostRef) {
        registerInstance(this, hostRef);
    }
    ProList.prototype.render = function () {
        return h("slot", null);
    };
    Object.defineProperty(ProList, "style", {
        get: function () { return ""; },
        enumerable: true,
        configurable: true
    });
    return ProList;
}());
var ProRadio = /** @class */ (function () {
    function ProRadio(hostRef) {
        var _this = this;
        registerInstance(this, hostRef);
        this.name = 'default-radio-name';
        this.disabled = false;
        this.checked = false;
        this.onClick = function () {
            if (_this.checked) {
                _this.proDeselect.emit();
            }
            else {
                _this.checked = true;
            }
        };
        this.proRadioDidLoad = createEvent(this, "proRadioDidLoad", 7);
        this.proRadioDidUnload = createEvent(this, "proRadioDidUnload", 7);
        this.proSelect = createEvent(this, "proSelect", 7);
        this.proDeselect = createEvent(this, "proDeselect", 7);
        this.proStyle = createEvent(this, "proStyle", 7);
    }
    ProRadio.prototype.nameChanged = function () {
        this.emitStyle();
    };
    ProRadio.prototype.checkedChanged = function (isChecked) {
        if (isChecked) {
            this.proSelect.emit({
                checked: true,
                value: this.value
            });
        }
        this.emitStyle();
    };
    ProRadio.prototype.disabledChanged = function () {
        this.emitStyle();
    };
    ProRadio.prototype.componentWillLoad = function () {
        this.emitStyle();
    };
    ProRadio.prototype.componentDidLoad = function () {
        this.proRadioDidLoad.emit();
    };
    ProRadio.prototype.componentDidUnload = function () {
        this.proRadioDidUnload.emit();
    };
    ProRadio.prototype.emitStyle = function () {
        this.proStyle.emit({
            'radio-checked': this.checked,
            'interactive-disabled': this.disabled,
            'radio-value': this.value
        });
    };
    /* <Host
                    onClick={this.onClick}
                    role="radio"
                    class={{
                        'radio-checked': this.checked,
                        'radio-disabled': this.disabled
                    }}
                >
                    <div class="radio-icon">
                        <div class="radio-inner" />
                    </div>
                    <button
                        type="button"
                        disabled={this.disabled}
                    >
                    </button>
                </Host> */
    ProRadio.prototype.render = function () {
        return (h("input", { type: "radio", onClick: this.onClick, name: this.name, value: this.value, checked: this.checked }));
    };
    Object.defineProperty(ProRadio.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ProRadio, "watchers", {
        get: function () {
            return {
                "name": ["nameChanged"],
                "checked": ["checkedChanged"],
                "disabled": ["disabledChanged"]
            };
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ProRadio, "style", {
        get: function () { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}input[type=radio]:after{background-color:#b7b7b7}input[type=radio]:after,input[type=radio]:checked:after{width:15px;height:15px;border-radius:15px;top:-2px;left:-1px;position:relative;content:\"\";display:inline-block;visibility:visible;border:2px solid #fff}input[type=radio]:checked:after{background-color:#30bced}input[type=radio]:hover:after{background:#1e96ff;-webkit-transition:.5s;transition:.5s}"; },
        enumerable: true,
        configurable: true
    });
    return ProRadio;
}());
var ProRadioGroup = /** @class */ (function () {
    function ProRadioGroup(hostRef) {
        registerInstance(this, hostRef);
        this.selects = [];
    }
    ProRadioGroup.prototype.onRadioDidLoad = function (ev) {
        var select = ev.target;
        select.name = this.name;
        this.selects.push(select);
        this.updateRadios();
    };
    ProRadioGroup.prototype.onRadioDidUnload = function (ev) {
        var index = this.selects.indexOf(ev.target);
        if (index > -1) {
            this.selects.splice(index, 1);
        }
    };
    ProRadioGroup.prototype.onSelect = function (ev) {
        var selectedRadio = ev.target;
        if (selectedRadio) {
            selectedRadio.checked = true;
            this.value = selectedRadio.value;
        }
        this.updateRadios();
    };
    ProRadioGroup.prototype.onDeselect = function (ev) {
        var selectedRadio = ev.target;
        if (selectedRadio) {
            selectedRadio.checked = false;
            this.value = undefined;
        }
        this.updateRadios();
    };
    ProRadioGroup.prototype.componentDidLoad = function () {
        this.updateRadios();
    };
    ProRadioGroup.prototype.updateRadios = function () {
        var value = this.value;
        var hasChecked = false;
        for (var _i = 0, _a = this.selects; _i < _a.length; _i++) {
            var select = _a[_i];
            if (!hasChecked && select.value === value) {
                hasChecked = true;
                select.checked = true;
            }
            else {
                select.checked = false;
            }
        }
    };
    ProRadioGroup.prototype.render = function () {
        return h("slot", null);
    };
    Object.defineProperty(ProRadioGroup.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ProRadioGroup, "style", {
        get: function () { return ""; },
        enumerable: true,
        configurable: true
    });
    return ProRadioGroup;
}());
export { ProButton as pro_button, ProCheckbox as pro_checkbox, ProInput as pro_input, ProList as pro_list, ProRadio as pro_radio, ProRadioGroup as pro_radio_group };
