// pro-ui-library/dist/esm/legacy/pro-card_4.entry.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
import { r as registerInstance, g as getContext, h, H as Host, c as getElement } from './chunk-c73a2de7.js';
var ProCard = /** @class */ (function () {
    function ProCard(hostRef) {
        registerInstance(this, hostRef);
        this.button = false;
        this.type = 'button';
        this.disabled = false;
        this.win = getContext(this, "window");
    }
    ProCard.prototype.isClickable = function () {
        return (this.href !== undefined || this.button);
    };
    ProCard.prototype.hostData = function () {
        return {
            class: {
                'card-disabled': this.disabled,
                'ion-activatable': this.isClickable()
            }
        };
    };
    ProCard.prototype.__stencil_render = function () {
        var clickable = this.isClickable();
        if (!clickable) {
            return [
                h("slot", null)
            ];
        }
        var href = this.href;
        var TagType = clickable ? (href === undefined ? 'button' : 'a') : 'div';
        var attrs = (TagType === 'button')
            ? { type: this.type }
            : {
                href: this.href,
                rel: this.rel,
                target: this.target
            };
        return (h(TagType, Object.assign({}, attrs, { class: "card-native", disabled: this.disabled }), h("slot", null), clickable && h("pro-ripple-effect", null)));
    };
    ProCard.prototype.render = function () { return h(Host, this.hostData(), this.__stencil_render()); };
    Object.defineProperty(ProCard, "style", {
        get: function () { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}pro-card{display:block;position:relative;overflow:hidden;-webkit-box-shadow:0 1px 3px #b7b7b7;box-shadow:0 1px 3px #b7b7b7;-webkit-transition:.3s;transition:.3s;border-color:#0a0908;border-width:1px;font-size:16px;padding:5px;margin:5px;background-color:#fdfffc;min-height:200px;max-height:500px}"; },
        enumerable: true,
        configurable: true
    });
    return ProCard;
}());
var ProImage = /** @class */ (function () {
    function ProImage(hostRef) {
        registerInstance(this, hostRef);
    }
    // @Watch() src
    ProImage.prototype.render = function () {
        return (h("img", { src: this.src, alt: this.alt, decoding: "async" }));
    };
    Object.defineProperty(ProImage, "style", {
        get: function () { return ":host{display:block;-o-object-fit:contain;object-fit:contain}:host img{display:block;min-width:50px;min-height:50px;max-width:150px;max-height:150px;width:100%;height:100%;-o-object-fit:inherit;object-fit:inherit;-o-object-position:inherit;object-position:inherit}"; },
        enumerable: true,
        configurable: true
    });
    return ProImage;
}());
var ProLabel = /** @class */ (function () {
    function ProLabel(hostRef) {
        registerInstance(this, hostRef);
    }
    ProLabel.prototype.render = function () {
        return (h("label", { htmlfor: this.for }, h("slot", null)));
    };
    Object.defineProperty(ProLabel, "style", {
        get: function () { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}pro-label{font-size:12px;font-style:italic;text-align:center}"; },
        enumerable: true,
        configurable: true
    });
    return ProLabel;
}());
var ProModal = /** @class */ (function () {
    function ProModal(hostRef) {
        registerInstance(this, hostRef);
        this.presented = false;
    }
    ProModal.prototype.present = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                if (!this.presented)
                    this.presented = true;
                return [2 /*return*/];
            });
        });
    };
    ProModal.prototype.dismiss = function () {
        return __awaiter(this, void 0, void 0, function () {
            return __generator(this, function (_a) {
                if (this.presented)
                    this.presented = false;
                return [2 /*return*/];
            });
        });
    };
    ProModal.prototype.render = function () {
        var _this = this;
        return (this.presented) ? (h(Host, null, h("pro-backdrop", { onClick: function () { return _this.dismiss(); } }), h("div", { class: "modal-window" }, h("div", { class: "modal-content" }, h("slot", null))))) : (h(Host, null));
    };
    Object.defineProperty(ProModal.prototype, "el", {
        get: function () { return getElement(this); },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(ProModal, "style", {
        get: function () { return "\@import url(\"https://fonts.googleapis.com/css?family=Raleway&display=swap\");h1,h2,h3,h4,h5,h6{font-weight:500;line-height:1.2}h1{font-size:30px}h2{font-size:28px}h3{font-size:26px}h4{font-size:24px}h5{font-size:22px}h6{font-size:18px}p{font-size:16px}.good-weather{background:#fcd581;background-image:-webkit-gradient(linear,left bottom,left top,from(#fcd581),color-stop(65%,#fdfffc));background-image:linear-gradient(0deg,#fcd581,#fdfffc 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#0a0908}.bad-weather{background:#0a0908;background:-webkit-gradient(linear,left bottom,left top,from(#0a0908),color-stop(65%,#263a3c));background:linear-gradient(0deg,#0a0908,#263a3c 65%);background-repeat:no-repeat;background-attachment:fixed;height:100%;color:#fdfffc}html{font-family:Raleway,Helvetica,sans-serif;font-size:100%;background-color:#fdfffc;color:#0a0908}#body,html{height:100%;width:100%}#body{background-repeat:no-repeat;background-attachment:fixed}:host{width:100%}\@media only screen and (max-width:600px){pro-modal{width:50%;height:50%}}:host .modal-window{position:absolute;right:0;left:0;top:2%;bottom:10%;margin-right:auto;margin-left:auto;z-index:15;background:#fdfffc;border-radius:25px;width:80%;min-height:40%;max-height:60%;padding:30px;-ms-flex-align:center;align-items:center;-ms-flex-pack:center;justify-content:center}\@media only screen and (max-width:600px){:host .modal-window{width:80%;min-height:50%}}\@media only screen and (min-width:600px){:host .modal-window{width:80%;min-height:50%}}:host .modal-content{width:100%;height:100%}"; },
        enumerable: true,
        configurable: true
    });
    return ProModal;
}());
export { ProCard as pro_card, ProImage as pro_image, ProLabel as pro_label, ProModal as pro_modal };
